lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1060,25 +1059,12 @@ readDelegation walletState = do
 -- we return the epoch of the node tip.
 getCurrentEpochSlotting
     :: HasCallStack => NetworkLayer IO block -> IO CurrentEpochSlotting
-getCurrentEpochSlotting nl = do
-    epoch <- getCurrentEpoch
-    mkCurrentEpochSlotting ti epoch
-  where
-    ti = Slotting.expectAndThrowFailures $ timeInterpreter nl
-
-    getCurrentEpoch =
-        currentEpochFromWallClock >>= \case
```

Comment:
Removing this is definitely more consistent. 👍

However, it may have an impact on Daedalus users: If the node has not fully synched yet, they will see an outdated delegation status. 🤔 But I think that this is ok — the network information endpoints tell user what `tip` the wallet sees, and it cannot reasonably look into the future. We may still want to record this as a "user-facing change", though.